tools/pubsys/src/repo.rs

```diff
@@ -430,30 +437,34 @@ pub(crate) fn run(args: &Args, repo_args: &RepoArgs) -> Result<()> {
 
     // Sign repo   =^..^=   =^..^=   =^..^=   =^..^=
 
-    let signing_key_config = infra_config
-        .signing_keys
-        .as_ref()
-        .context(error::MissingConfig {
-            missing: "signing_keys",
-        })?
-        .get(&repo_args.signing_key)
-        .context(error::MissingConfig {
-            missing: format!("profile {} in signing_keys", &repo_args.signing_key),
-        })?;
+    // Check if we have a signing key defined in Infra.toml; if not, we'll fall back to the
+    // generated local key.
+    let signing_key_config = repo_config.signing_keys.as_ref();
 
     let key_source: Box<dyn KeySource> = match signing_key_config {
-        SigningKeyConfig::file { path } => Box::new(LocalKeySource { path: path.clone() }),
-        SigningKeyConfig::kms { key_id } => Box::new(KmsKeySource {
+        Some(SigningKeyConfig::file { path }) => Box::new(LocalKeySource { path: path.clone() }),
+        Some(SigningKeyConfig::kms { key_id }) => Box::new(KmsKeySource {
             profile: None,
             key_id: key_id.clone(),
             client: None,
             signing_algorithm: KmsSigningAlgorithm::RsassaPssSha256,
         }),
-        SigningKeyConfig::ssm { parameter } => Box::new(SsmKeySource {
+        Some(SigningKeyConfig::ssm { parameter }) => Box::new(SsmKeySource {
             profile: None,
             parameter_name: parameter.clone(),
             key_id: None,
         }),
+        None => {
+            ensure!(
+                repo_args.default_key_path.exists(),
+                error::MissingConfig {
+                    missing: "signing_keys in repo config, and we found no local key",
```

Comment:
Should this be "no signing_keys in repo config..."?

Reply:
I don't think so, because we're saying what's missing, not what's present... "*missing*: signing_keys ..."